src/relative.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2012-2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Taken from the pathdiff crate, which adapted the original rustc's path_relative_from
+// https://github.com/Manishearth/pathdiff/blob/master/src/lib.rs
+// https://github.com/rust-lang/rust/blob/e1d0de82cc40b666b88d4a6d2c9dcbc81d7ed27f/src/librustc_back/rpath.rs#L116-L158
+
+use std::path::*;
+
+/// Provides helper methods on Path and PathBuf to for creating RelativePath.
+pub trait PathExt {
```

Comment:
Please seal this trait, we don't want to allow downstream implementations so that it can evolve without breaking releases.